src/data/setoid/partition.lean

```diff
@@ -237,6 +244,35 @@ _ (subtype (@is_partition α)) _ (partial_order.to_preorder _) $ partition.order
 
 end partition
 
+section finpartition
+
+open_locale classical
+
+/-
+-- Inserted in data.finset.lattice.lean, line 1115
+lemma finset.sup_id_eq_sUnion {α} (s : finset (set α)) :
+  s.sup id = ⋃₀(↑s) :=
+finset.sup_id_eq_Sup _
+-/
+
+/-- A finite setoid partition furnishes a finpartition -/
+def finpartition.of_partition {c : finset (set α)}
+  (hc : setoid.is_partition (c : set (set α))) : finpartition (set.univ : set α) :=
+{ parts := c, --  finset (set α)
+  sup_indep := finset.sup_indep_iff_pairwise_disjoint.mpr $ eqv_classes_disjoint hc.2,
+  --  self.parts.sup_indep id
+  sup_parts := c.sup_id_eq_sUnion.trans hc.sUnion_eq_univ,
+  --  self.parts.sup id = (set.univ : set α)
+  not_bot_mem := hc.left --  ⊥ ∉ self.parts
+}
+
+/-- A finpartition gives rise to a setoid partition -/
+theorem is_partition.of_finpartition(f : finpartition (set.univ : set α)) :
+  setoid.is_partition (f.parts : set (set α)) :=
```

Comment:
For dot notation"
```suggestion
theorem finpartition.is_partition_parts (f : finpartition (set.univ : set α)) :
  setoid.is_partition (f.parts : set (set α)) :=
```

Reply:
Done

Reply:
Doesn't look like you renamed it as I suggested to `finpartition.is_partition_parts`